FHEM/00_SIGNALduino.pm

```diff
@@ -4231,42 +4231,49 @@ sub SIGNALduino_FW_getProtocolList
 	
 	my $id;
 	my $ret;
-	my $s;
-	my $moduleId;
+	my $devFlag = 0;	# 1 - develop version
+	my $devText = "";
+	my %mIdListH = ();
 	my @IdList = ();
 	
+	my $whitelist = AttrVal($name,"whitelist_IDs","#");
+	
+	my $develop = AttrVal($name,"development","");
+	if (length($develop) > 0 && ($develop eq "1" || substr($develop,0,1) eq "y")) {		# develop version
+		$devFlag = 1;
+		$devText = "development version - ";
+	}
+	
+	my $mIdList = join(",",@{$hash->{msIdList}},@{$hash->{muIdList}},@{$hash->{mcIdList}});
+	%mIdListH = map { $_ => 1 } split(",", $mIdList);
+	SIGNALduino_Log3 $name,4, "$name IdList: $mIdList";
+	
 	foreach $id (keys %ProtocolListSIGNALduino)
 	{
-		next if ($id eq 'id');
+		next if ($id eq 'id' || $id >= 900);
```

Comment:
wieso >900 überspringen? Ist das gut?

Reply:
Dies ist eine Vorbereitung für einen weiteren Patch mit dem die Version der ProtocolList in den Internals angezeigt wird.

```
	"999" =>  # 
	{
		versionProtocolList  => 'dev_17.12.18'
	}
```

Reply:
Können wir dem Key nicht den Namen  "Version" geben? GGf. müssten wir den halt beim einlesen filtern.

Reply:
Ich finde es besser, wenn wir bei den keys nur Zahlen haben.

Reply:
Da es in " steht, ist es ohnehin ein String.

Reply:
Wir haben bis jetzt nur Zahlen als IDs, wenn wir das vermischen ist es denkbar, daß uns diese unsauberheit irgendwann mal einholt.
Das kann raus ` if ($id eq 'id');` inzwischen ist kein "id" mehr enthalten. Evtl war das mal durch ein unsauberes if (defined drin, z.B. `if (defined(ProtocolListSIGNALduino{id}{xyz})`

Reply:
Der Protokoll Hash wird ja in `SIGNALduino_LoadProtocolHash `eingelesen. Dort würde ich  das direkt abfangen